src/core/DataFile.cpp

```diff
@@ -1355,6 +1355,35 @@ void DataFile::upgrade_1_3_0()
 	}
 }
 
+void DataFile::upgrade_noHiddenClipNames()
+{
+	QDomNodeList tracks = elementsByTagName("track");
+
+	auto clearDefaultNames = [](QDomNodeList clips, QString trackName)
+	{
+		for (int j = 0; j < clips.size(); ++j)
+		{
+			QDomElement clip = clips.item(j).toElement();
+			QString clipName = clip.attribute("name", "");
+			if (clipName == trackName) { clip.setAttribute("name", ""); }
+		}
+	};
+
+	for (int i = 0; i < tracks.size(); ++i)
+	{
+		QDomElement track = tracks.item(i).toElement();
+		QString trackName = track.attribute("name", "");
+
+		QDomNodeList instClips = elementsByTagName("pattern");
+		QDomNodeList autoClips = elementsByTagName("automationpattern");
+		QDomNodeList bbClips = elementsByTagName("bbtco");
```

Comment:
Shouldn't these be `track.elementsByTagName`? Otherwise a clip's name will be cleared if it matches the name of any track in the project, not just its parent track.

Reply:
Oh dear, good catch. Will update, probably later today.

Reply:
Addressed in latest commit. I also went a bit out of scope with said commit, so feel free to tell me if I should revert the out-of-scope parts.